chromadb/test/test_api.py

```diff
@@ -676,6 +676,53 @@ def test_where_ne_eq_number(api):
     assert len(items["metadatas"]) == 1
 
 
+def test_where_like_string(api):
+    api.reset()
+    collection = api.create_collection("test_where_like")
+    collection.add(**operator_records)
+    items = collection.get(where={"string_value": {"$like": "%wo%"}})
+    assert len(items["metadatas"]) == 1
+    items = collection.get(where={"string_value": {"$like": "%o%"}})
+    assert len(items["metadatas"]) == 2
+    items = collection.get(where={"float_value": {"$like": "%o%"}})
+    assert len(items["metadatas"]) == 0
+    items = collection.get(where={"string_value": {"$nlike": "%o%"}})
+    assert len(items["metadatas"]) == 0
+
+
+def test_where_nlike_string(api):
+    api.reset()
+    collection = api.create_collection("test_where_like")
+    collection.add(**operator_records)
+    items = collection.get(where={"string_value": {"$nlike": "%wo%"}})
+    assert len(items["metadatas"]) == 1
+    items = collection.get(where={"string_value": {"$nlike": "%o%"}})
+    assert len(items["metadatas"]) == 0
+    items = collection.get(where={"string_value": {"$nlike": "%h%"}})
+    assert len(items["metadatas"]) == 2
+    items = collection.get(where={"float_value": {"$nlike": "%wo%"}})
+    assert len(items["metadatas"]) == 0
+
+
+def test_where_valid_operators_like(api):
+    api.reset()
+    collection = api.create_collection("test_where_valid_like_operators")
+    collection.add(**operator_records)
+
+    with pytest.raises(ValueError):
+        collection.get(where={"b": {"$like": 4}})  # invalid
+    with pytest.raises(ValueError):
+        collection.get(
+            where={
+                "$or": [
+                    {"a": {"$like": "first"}},  # invalid
```

Comment:
Are we saying partial/exact matches are unsupported?

Partial - `SELECT * FROM products WHERE description LIKE 'monitor';`
Exact  - `SELECT * FROM customer WHERE city LIKE 'New York';`

Reply:
partial matches are supported, they just need wildcard characters.
Eg, `SELECT * FROM products WHERE description LIKE 'monitor';` is considered an exact match, 
Eg, `SELECT * FROM products WHERE description LIKE '%monitor%';` is considered an exact match,

the comments should really say:
```
collection.get(
            where={
                "$or": [
                    {"a": {"$like": "first"}},  # valid
                    {"b": {"$like": 4}},  # invalid
                    {"$like": "second"},  # valid
                ]
            }
        )
```

Although in the case of the last one, it's actually trying to match the '$like' operator as a metadata key, the equivalent to:
`{'$like':{'$eq','second'}}`. 

This happens for any operator that isn't `$and`, `$or`, `$in`, or `$nin`, I don't know if this is supposed to be intended given the other tests.